tests/Functional/HStringMapTest.php

```diff
@@ -39,4 +40,24 @@ public function testStringMapRot13()
         $expected = "sbbone";
         $this->assertEquals($expected, $newString);
     }
+
+    public function testStringWithVariadicArray()
+    {
+        $myArray = new HArray([
+            "apple",
+            "butter",
+            "cookie",
+            "donut",
+            "eclaire",
+            "framboise",
+        ]);
+
+        $oddPairing = function ($sentenceWord, $arrayWord) {
+            return sprintf("-%s:%s-", $sentenceWord, $arrayWord);
+        };
+
+        $pairs = $this->aString->map($oddPairing, $myArray);
+        $expected = "-f:apple--o:butter--o:cookie--b:donut--a:eclaire--r:framboise-";
```

Comment:
When implementing [array_zip()](https://github.com/johnkary/rethinkingloops/blob/master/examples/array_zip.php#L48) I had to account for cases where the two arrays differed in length. This includes:

1. $this->aString length shorter than $myArray
2. $this->aString length same length as $myArray
3. $this->aString length longer than $myArray

Perhaps see how array_map() works in those cases and mimic it?